extensions/amp-render/1.0/amp-render.js

```diff
@@ -212,7 +214,51 @@ export class AmpRender extends BaseElement {
       },
       'onReady': () => {
         this.toggleLoading(false);
-        this.togglePlaceholder(false);
+        if (this.element.getAttribute('layout') !== Layout.CONTAINER) {
+          this.togglePlaceholder(false);
+          return;
+        }
+        if (!this.getPlaceholder()) {
+          user().error(TAG, 'placeholder required with layout="container"');
+          return;
+        }
```

Comment:
This block can be executed once in `isLayoutSupported`. `layout` attr is AMP specific and therefore not mutable, so check only needs to execute once.

The condition + error can also be expressed via `userAssert` instead.